common/materialContainer/MaterialContainer.h

```diff
@@ -76,7 +76,7 @@ template <class T> class MaterialContainer
     , contentFreq(MC.contentFreq)
   {
   }
-  MaterialContainer& operator=(const MaterialContainer& mc) { return *this; }
+  MaterialContainer& operator=(const MaterialContainer&) { return *this; }
```

Comment:
We should not be using stack objects in vectors. Either used a shared_ptr or just a regular pointer and you do not need this definition.